doc/eval.jax

```diff
@@ -5281,6 +5292,16 @@ getcharsearch()						*getcharsearch()*
 			:nnoremap <expr> , getcharsearch().forward ? ',' : ';'
 <		|setcharsearch()|も参照。
 
+getcharstr([expr])					*getcharstr()*
+		ユーザーまたは入力ストリームから1文字を文字列として取得する。
+		[expr] が省略されたときは1文字を取得できるまで待つ。
+		[expr] が0もしくは偽のときは1文字を取得できる場合のみ取得する。
+		取得できなければ空文字列を返す。
+		[expr] が1もしくは真のときは1文字を取得できるか判定し、実際に
+		は取得しない。取得できないときは空文字列を返す。
+		結果の数値が文字列に変換される以外は |getchar()| と同様に動作
+		する。
```

Comment:
`Otherwise` を訳したほうが良いと思います。

```suggestion
		それ以外のときは |getchar()| と同様に動作する。ただし数値の結
		果は文字列に変換される。
```

Reply:
うーーん
イディオムというわけじゃないみたいなので、たぶんそうするのが翻訳としては正しい気がしますが
「それ以外」が前の文字列変換ルール説明でそれを受けつつ、後段のexpectで例外として再度説明するとちょっと変かなと思います。

```
これはxxです...
0なら...
1なら...
これ以外はgetcharに同じです(ただし数値は文字列に変換される == 0と1のときのこと)
```

なので、Otherwiseがexpectのことを指すのと同義でもあるので、

```
これはxxです...
0なら...
1なら...
数値は文字列に変換される(これら == 0と1のときのこと)という以外はgetcharに同じです
```

が読みやすいかなあと思います。

@h-east さん、他の方
どう思いますか?

Reply:
> 数値は文字列に変換される == 0と1のときのこと

んー、そういう意味ではない気がします。

1. 引数 `[expr]` が省略されたときは A という挙動になる
2. 引数 `[expr]` が 0 か false のときは B という挙動になる
3. 引数 `[expr]` が 1 か true のときは Cという挙動になる
4. それ以外のとき、即ち以上 1-3 のいずれも満たさないときは D という挙動になる

という構造になっていて、挙動 D は同条件下 (1-3 のいずれも満たさないという条件) で `getchar()` を使った場合と同様だが、`getchar()` で戻り値が数値となるような (さらに細分化された) 条件下ではその数値を対応する文字列に変換したものが (`getcharstr()`の場合の) 戻り値になる、と言っているはずです (`getcharstr()` は常に文字列を返すので)。

Reply:
「1-3 のいずれも満たさない」パターンってないと思います。
それはさておき、ニュアンス的には、「上にいろいろ説明したけど要は、結果の数値が文字列に変換されること以外 getchar() と同じだよ」って感じだと思うので、tsuyoshichoさんの訳でいいんじゃないでしょうか。

Reply:
今回は一応キープということで

Reply:
えっ、文字列とか辞書とかリストの場合は 1-3 のいずれも満たさないのでは… とはいえソースコードを見ると確かにそんな感じの実装になっていますね…失礼しました (そもそも辞書とかを渡すとエラーになるのに気づいていなかった)
`Otherwise` を `要するに` みたいな意味で使っているとしたらかなり不適切な英語に思えます。あと翻訳にも `要するに` みたいな文言を入れたほうが読みやすいように感じますが、どうでしょうか?

Reply:
なんというか Otherwise の文がくどいのがよくないんだろうなと。
expect の節なしで、「それ以外はgetcharと同じ」だけでも説明は十分なはずだし...

原文にどこまで忠実で、どれだけ意訳する/修正を考えて送るか程度が難しいですね..ただ、「それ以外は全部同じ」は伝えなきゃいけないけど、別に前の節を要約するわけではないので、そっちの意味合いは薄そうなので、このままでいいかな...

Reply:
意味的には無くすなら `except` 節ではなく `Otherwise` な気はしますね。
もしかして `In other words` を `Otherwise` に typo したのかな…? (あまりに大胆
個人的には ↓ とするのが好みですが、おまかせします 🙏 

```suggestion
		要するに数値の結果が文字列に変換される以外は |getchar()| と同
		じである。
```

Reply:
大胆 typo 説を提唱しておきました https://github.com/vim-jp/issues/issues/1321#issuecomment-872363792

Reply:
この patch は取り込まないと決めた場合も "Resolve conversation" しておいていただけると… そしたら  approve しますので 🙇 (このままだと merge しづらいのかもというあれです)